engine/execution/state/delta/view.go

```diff
@@ -119,32 +117,37 @@ func (v *View) RegisterUpdates() ([]flow.RegisterID, []flow.RegisterValue) {
 // This function will return an error if it fails to read from the underlying
 // data source for this view.
 func (v *View) Get(owner, controller, key string) (flow.RegisterValue, error) {
+	registerID := flow.NewRegisterID(owner, controller, key)
+
 	value, exists := v.delta.Get(owner, controller, key)
 	if exists {
-		// every time we read a value (order preserving) we update spock
-		var err error
-		if value != nil {
-			err = v.updateSpock(value)
+		// every time we read a value (order preserving) we update the secret
+		// with the registerID only (value is not required)
+		err := v.updateSpockSecret(registerID.Bytes())
+		if err != nil {
+			return nil, fmt.Errorf("get register failed: %w", err)
 		}
-		return value, err
+		return value, nil
 	}
 
 	value, err := v.readFunc(owner, controller, key)
 	if err != nil {
```

Comment:
If this read fails, should we nonetheless attempt to update the secret with the attempted read?

Reply:
Good catch! 

@m4ksio what is the difference between the read call [here](https://github.com/onflow/flow-go/blob/master/engine/execution/state/delta/view.go#L122) and the other one [here](https://github.com/onflow/flow-go/blob/master/engine/execution/state/delta/view.go#L132)? and if the call to `readFunc` fails, is it expected that it will fail in all ENs?

Reply:
We first check if the value is in local delta, if not, we use `readFunc` which is usually a read from parent delta, which in turns read from its parent delta up to the actual state trie.

Any error while reading should be considered fatal so I would say updating spock doesn't matter, as it should be discarded anyway

Reply:
Thanks!
Will `readFunc` error if the register is not found (which I guess is a possible not fatal case) ?  I can see `delta.Get` has an `exist` return but `ReadFunc` doesn't.

Reply:
> Will readFunc error if the register is not found (which I guess is a possible not fatal case) ?

No. Empty slice/nil (well zero-length slice) is assumed to mean does not exists, ie. you can't store zero-length array.

> I can see delta.Get has an exist return but ReadFunc doesn't.

`delta.Get` accesses map and this cannot error in Go, but its important to distinguish if there was a response (ie. nil was stored) vs no information (defer to asking `ReadFunc`).

Reply:
ah I understand, that makes sense!